src/patch/parse.js

```diff
@@ -7,41 +7,38 @@ export function parsePatch(uniDiff, options = {}) {
     let index = {};
     list.push(index);
 
-    // Ignore any leading junk
-    while (i < diffstr.length) {
-      if (/^(Index:|diff -r|@@)/.test(diffstr[i])) {
+    // Parse diff metadata
+    for (; i < diffstr.length; i++) {
```

Comment:
"This is to mimic style" means it's stylistic. I'm glad to discuss stylistic changes in another PR if you want to spend your time on that but I don't accept them for functional changes.